Digipost/OAuthToken.swift

```diff
@@ -37,10 +37,85 @@ class OAuthToken: NSObject, NSCoding, DebugPrintable, Printable{
         }
     }
 
-    var expires : NSDate?
+    var expires : NSDate
 
     var scope: String?
 
+    init(expiryDate: NSDate) {
+        self.expires = expiryDate
+        super.init()
+    }
+
+    required convenience init(coder decoder: NSCoder) {
+        // If token is archived without expirydate, for example if upgrading from an older client, set expirydate to now(), to force reauthentication
+        let expires : NSDate = {
+            if let expiryDate =  decoder.decodeObjectForKey(Keys.expiresKey) as? NSDate {
+                return expiryDate
+            } else {
+                return NSDate()
+            }
+            }()
+        self.init(expiryDate: expires)
+        self.scope = decoder.decodeObjectForKey(Keys.scopeKey) as! String!
+        self.refreshToken = decoder.decodeObjectForKey(Keys.refreshTokenKey) as? String
+        self.accessToken = decoder.decodeObjectForKey(Keys.accessTokenKey) as? String
+    }
+
+    // Initializer used when migrating from single scope to multiple scopes - version 2.3
+    convenience init?(refreshToken: String?, scope: String) {
+        self.init(expiryDate:NSDate())
+
+        if let acutalRefreshToken = refreshToken as String? {
+            self.refreshToken = acutalRefreshToken
+        } else {
+            return nil
+        }
+
+        self.scope = scope
+        storeInKeyChain()
+    }
+
+    convenience init?(refreshToken: String?, accessToken: String?, scope: String, expiresInSeconds: NSNumber) {
```

Comment:
Could these initializers call each others? Right now we duplicate quite a lot of functionality between them.